### DISCUSSION TITLE
Add support for SysML

### DISCUSSION THREAD

**Discussion by kuwv** [networkx/networkx#6632] [Ideas]
Currently, networkx supports GraphML but SysML is also widely used. This could allow networkx to be used by a larger group of client / server applications.

**Comment by rossbar** [comment]
Is SysML related to networks? I couldn't get a sense of what types of data are represented from a quick scan of the latest standard published [on the sysml specs page](https://sysml.org/sysml-specs/)

**Comment by kuwv** [comment]
@rossbar SysML is most widely used for systems engineering. That would include networking, "systems and systems-of-systems", which is used in aerospace and automotive.

https://en.wikipedia.org/wiki/Systems_Modeling_Language

**Comment by dschult** [comment]
The link to wikipedia doesn't help me see that this is related to networks. The pictures show diagrams that could be called directed networks. But the content seems to be less about doing network science and more about network drawing. But I haven't looked closely.  We're trying to weigh the cost of additional maintenance keeping yet-another-graph-markdown-language supported.

Thoughts?

**Reply by kuwv** [reply]
Yeah, I understand that. I provided the wiki intending to link the 9 types of diagrams but just sent the whole article instead.

This paper discussing graph theory with SysML might help:
http://gvsets.ndia-mich.org/documents/SE/2015/Understanding%20Systems%20through%20Graph%20Theory%20and%20Dynamic%20Visualization.pdf

Also, Cisco uses SysML so I don't know if that is any interest.
https://blogs.cisco.com/cloud/cloud-native

**Comment by rossbar** [comment]
> SysML is most widely used for systems engineering.

Thanks for the info. I too have concerns about maintenance burden, and it doesn't strike me as the cleanest fit. GraphML makes sense because the format is specifically designed for graphs/networks. I certainly understand that networks are an important component in systems design, but I suspect the systems domain comprises a lot of other things as well.

I have no idea what the systems design landscape looks like in Python - a quick search seems to indicate that [Gaphor](https://gaphor.org/) is a go-to library. Maybe it would make sense to add `to/from_networkx` functions there?

**Reply by kuwv** [reply]
I brought up a separate topic to the Graphor developers. They suggested separating their graph capability into a library. That might actually work here if they do that.

https://github.com/gaphor/gaphor/issues/2223

**Comment by kuwv** [comment]
Maybe a little more info would be helpful.

SysML is used by MBSE development to replace the documentation required by Waterfall process with diagrams. These diagrams are typically a holistic interconnected view (usually TOGAF, DoDAF, or UAFML) of multiple layers of these diagrams and serve multiple purposes. The main capability of SysML aside from inter-connectivity is the ability to be processed with mathematical proofs in software-in-the-loop and hardware-in-the-loop simulations.

The reason why this would be valuable to `networkx` and developers is that it would provide the capability to dynamically generate these graph diagrams. These diagrams can then be run from simulators and used by other diagram as their basis for structural or behavioral design. It is also NIST standard so it pretty much makes it a USG standard too.

https://docs.aws.amazon.com/whitepapers/latest/model-based-systems-engineering/what-is-mbse-and-why-do-industries-start-to-use.html
